### PR TITLE
add URLMap middleware

### PR DESCRIPTION
**Pull request by mattn**

### CODE REVIEW

File: lib/Crust/App/URLMap.pm6 [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        my $res = $map<app>($env);
+        $env<PATH_INFO> = $orig_path_info;
+        $env<SCRIPT_NAME> = $orig_script_name;
+        return $res;
+    }
+
+    return [404, ['Content-Type' => 'text/plain'], ["Not Found".encode('ascii')]];
+}
+
+# TODO inherit Crust::Component
```
Comment (moznion) [inline lib/Crust/App/URLMap.pm6]: Looks not necessary

File: lib/Crust/App/URLMap.pm6 [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+multi method map(Str $location, Callable $callable) {
+    my $loc = $location;
+    my Str $host;
+    if $loc ~~ /^ 'http' 's'? '://' (.*?) ('/' .*)/ {
+        $host = $0.Str || '';
+        $loc  = $1.Str || '';
+    } 
+    $loc = $loc.subst(/\/+ $/, '');
+    $!mapping.push: {host => $host, loc => $loc, app => $callable};
+}
```
Comment (tokuhirom) [inline lib/Crust/App/URLMap.pm6]: if it returns `self`, we can use it in fluent interface.

Comment (mattn) [inline lib/Crust/App/URLMap.pm6]: make sense

Comment (mattn) [inline lib/Crust/App/URLMap.pm6]: will add test

**Pull request merged**